src/libraries/Microsoft.PowerFx.Core/Parser/TexlParser.cs

```diff
@@ -95,6 +98,23 @@ public static ParseUserDefinitionResult ParseUserDefinitionScript(string script,
             return parser.ParseUDFsAndNamedFormulas(script, parserOptions);
         }
 
+        /// <summary>
+        /// Parses a script with user definitions. Same as <see cref="ParseUserDefinitionScript(string, ParserOptions)"/>.
+        /// but internal for prototyping of attribute support. 
+        /// </summary>
+        /// <param name="script">Script to be parsed.</param>
+        /// <param name="parserOptions">Options for parsing an expression.</param>
+        /// <returns><see cref="ParseUserDefinitionResult"/>.</returns>
+        internal static ParseUserDefinitionResult ParseUserDefinitionScriptWithAttributes(string script, ParserOptions parserOptions)
+        {
+            Contracts.AssertValue(parserOptions);
+            var flags = (Flags.NamedFormulas | Flags.AllowAttributes | (parserOptions.NumberIsFloat ? Flags.NumberIsFloat : 0)) | (parserOptions.AllowParseAsTypeLiteral ? Flags.AllowTypeLiteral : 0);

```

Comment:
>(Flags.NamedFormulas | Flags.AllowAttributes | (parserOptions.NumberIsFloat ? Flags.NumberIsFloat : 0)) | (parserOptions.AllowParseAsTypeLiteral ? Flags.AllowTypeLiteral : 0); [](http://example.com/codeflow?start=24&length=175)

This is almost the same logic as line 94; can it be refactored into a common place?